substrate/frame/identity/src/lib.rs

```diff
@@ -595,7 +578,8 @@ pub mod pallet {
 			let err_amount = T::Currency::unreserve(&sender, fee);
 			debug_assert!(err_amount.is_zero());
 			let judgements = id.judgements.len();
-			let extra_fields = id.info.additional.len();
+			#[allow(deprecated)]
+			let extra_fields = id.info.additional();
```

Comment:
This makes absolutely no sense. The benchmarking in this case is wrong. `MaxAdditionFields` is not changing the weight in any way (besides more pov size, but this needs to be addressed differently)

Reply:
Looking at it now, I agree that the benchmarking here is wrong because the function doesn't iterate through any additional fields. This applies to all other extrinsics too because the `additional` field of `IdentityInfo` is only constructed and read off-chain. Knowing the length of `additional` is important only for the storage deposit.

However, the change you referenced is purely a refactor. As you can see [here](https://github.com/paritytech/polkadot-sdk/blob/35eb133baab93d3f2f179df216b2cc175d7dcaf2/substrate/frame/identity/src/lib.rs#L598C48-L598C48), functionally it is the same as before this PR. I agree that we need to fix the weights though.

Reply:
The benchmarking was wrong before and that should obviously be addressed. But this PR does nothing to change this behavior getting into a release. Reverting it would be the same.